observation/reader_test.go

```diff
@@ -0,0 +1,170 @@
+package observation_test
+
+import (
+	"github.com/ONSdigital/dp-dataset-exporter/observation"
+	"github.com/ONSdigital/dp-dataset-exporter/observation/observationtest"
+	. "github.com/smartystreets/goconvey/convey"
+	"io"
+	"reflect"
+	"testing"
+)
+
+func TestReader_Read(t *testing.T) {
+
+	Convey("Given a reader with a mock CSV row reader that returns a single row", t, func() {
+
+		dummyRowContent := "csv,row,content"
+
+		mockRowReader := &observationtest.CSVRowReaderMock{
+			ReadFunc: func() (string, error) {
+				return dummyRowContent, nil
+			},
+		}
+
+		reader := observation.NewReader(mockRowReader)
```

Comment:
In the test it serves no purpose as it would call close on the mock. I would have to implement a close method on the mock so that it can be called from the test 😄